RecoBTag/FeatureTools/src/SeedingTracksConverter.cc

```diff
@@ -0,0 +1,232 @@
+
+#include "DataFormats/PatCandidates/interface/Jet.h"
+
+#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
+#include "TrackingTools/IPTools/interface/IPTools.h"
+#include "TrackingTools/Records/interface/TransientTrackRecord.h"
+#include "RecoVertex/VertexTools/interface/VertexDistance3D.h"
+
+#include "RecoBTag/TrackProbability/interface/HistogramProbabilityEstimator.h"
+
+#include "RecoBTag/FeatureTools/interface/SeedingTracksConverter.h"
+
+
+namespace btagbtvdeep {
+    
+        void seedingTracksToFeatures(edm::Handle<edm::View<reco::Candidate> > tracks,
+                                        const reco::Jet & jet,
+                                        const reco::Vertex & pv,                                            
+                                        edm::ESHandle<TransientTrackBuilder> & track_builder,
+                                        HistogramProbabilityEstimator* probabilityEstimator,
+                                        bool computeProbabilities,
+                                        std::vector<btagbtvdeep::SeedingTrackFeatures> & seedingT_features_vector
+                                        ) 
+        {
+            
+            GlobalVector jetdirection(jet.px(),jet.py(),jet.pz());
+            GlobalPoint pvp(pv.x(),pv.y(),pv.z());
+                        
+            std::vector<reco::TransientTrack> selectedTracks;          
+            std::vector<float> masses;
+                
+            for(typename edm::View<reco::Candidate>::const_iterator track = tracks->begin(); track != tracks->end(); ++track) {
+                
+                unsigned int k = track - tracks->begin();
+                
+                if(track->bestTrack() != nullptr && track->pt()>0.5) 
+                { 
+
+                    if (std::fabs(track->vz()-pv.z())<0.5)
+                    {
+                        selectedTracks.push_back(track_builder->build(tracks->ptrAt(k)));
+                        masses.push_back(track->mass());
+                                                
+                    }
+                    
+                }
+                
+            }
+                
+            std::multimap<double,std::pair<btagbtvdeep::SeedingTrackInfoBuilder,std::vector<btagbtvdeep::TrackPairFeatures>>> sortedSeedsMap;
+            std::multimap<double,btagbtvdeep::TrackPairInfoBuilder> sortedNeighboursMap;
+            
+            std::vector<btagbtvdeep::TrackPairFeatures> tp_features_vector;            
+            
+            sortedSeedsMap.clear();
+            seedingT_features_vector.clear();
+            
+            unsigned int selTrackCount=0;
+
+            for(auto const& it : selectedTracks){
+                
+                selTrackCount+=1;
+                sortedNeighboursMap.clear();
+                tp_features_vector.clear();        
+                
+                if (reco::deltaR(it.track(), jet) > 0.4) continue;
```

Comment:
from https://github.com/cms-sw/cmssw/pull/26088#discussion_r280985312

> a similar check better be also added to the loop that defines the selected tracks
> so that we do not keep checking all tracks from a PV for a given jet.